<launch>
    <!-- ORB-SLAM3 -->
    <node name="ros_wrapper_stereo_inertial" pkg="ros_wrapper" type="ros_wrapper_stereo_inertial" output="screen">
        <!-- for EuRoC dataset -->
        <remap from="/camera/left/image_raw"    to="/cam0/image_raw"/>
        <remap from="/camera/right/image_raw"   to="/cam1/image_raw"/>
        <remap from="/imu"                      to="/imu0"/>

        <!-- Parameters for original ORB-SLAM3 -->
        <param name="settings_file" type="string"   value="$(find ros_wrapper)/config/rs_d455/estimator_config.yaml" />

        <!-- Parameters for ROS -->
        <param name="world_frame_id"    type="string"   value="world" />
        <param name="cam_frame_id"      type="string"   value="camera" />
        <param name="enable_pangolin"   type="bool"     value="true" />
    </node>

    <!-- Visualization - RViz-->
    <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find ros_wrapper)/config/orb_slam3_with_imu.rviz" output="screen" /> -->

    <!-- <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_orb_slam3" output="screen" ns="orb_slam3_ros" >
        <param name="/target_frame_name" value="/world" />
        <param name="/source_frame_name" value="/camera" />
        <param name="/trajectory_update_rate" value="20.0" />
        <param name="/trajectory_publish_rate" value="20.0" />
    </node> -->
</launch>